bundles/test/org.eclipse.smarthome.test/src/main/java/org/eclipse/smarthome/test/java/JavaOSGiTest.java

```diff
@@ -200,10 +199,19 @@ private BundleContext initBundleContext() {
             final Dictionary<String, ?> properties) {
         assertThat(interfaceName, is(notNullValue()));
         final ServiceRegistration<?> srvReg = bundleContext.registerService(interfaceName, service, properties);
-        registeredServices.put(interfaceName, srvReg);
+        saveServiceRegistration(interfaceName, srvReg);
         return srvReg;
     }
 
+    private void saveServiceRegistration(final String interfaceName, final ServiceRegistration<?> srvReg) {
+        List<ServiceRegistration<?>> regs = registeredServices.get(interfaceName);
+        if (regs == null) {
+            regs = new ArrayList<>();
+        }
+        regs.add(srvReg);
+        registeredServices.put(interfaceName, regs);
```

Comment:
This should be moved to the if branch. A new entry needs only to be pushed if regs is null.